solvers/ChambollePockPDsplit.py

```diff
@@ -67,3 +67,14 @@ def _prox_huber(self, u, mu):
             u / (mu + 1.0),
             u - self.delta * mu * np.sign(u),
         )
+
+    def get_l2norm(self, A, n_iter=100):
+        if isinstance(A, np.ndarray):
+            return np.linalg.norm(A, ord=2)
+        else:
+            AtA = A.T @ A
+            x = np.random.randn(AtA.shape[0])
+            for _ in range(n_iter):
+                x = AtA @ x
```

Comment:
@EnLAI111 can you check if doing `(A.T @ A) @ x` is faster than `A.T @ (A @ x)`
in which case you should not create AtA